scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -18,45 +18,66 @@
   ; TODO: Should do lock sharding based on hash of pod-name.
   lock-object)
 
-(defn expected-state-equivalent?
+(defn cook-expected-state-equivalent?
   "Is the old and new state equivalent?"
   [old-state new-state]
   (= old-state new-state); TODO
   )
 
-(defn existing-state-equivalent?
+(defn k8s-actual-state-equivalent?
   "Is the old and new state equivalent?"
   [old-state new-state]
   ; TODO:
   ; This is not right. We need to tune/tweak this to
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-pod
+  "Kill pod is the same as deleting a pod. I semantically distinguish them. Delete is used for completed pods that
+  we're done with. Kill is used for possibly running pods we want to kill so that they fail. Returns a new
+  cook expected state dict of nil."
+  [api-client pod]
+  (api/delete-pod api-client pod)
+  nil)
 
-(defn kill-task
-  [api-client expected-state-dict pod]
-  (api/kill-task api-client pod)
-  expected-state-dict)
+(defn kill-pod
+  "Kill pod is the same as deleting a pod. I semantically distinguish them. Delete is used for completed pods that
+  we're done with. Kill is used for possibly running pods we want to kill so that they fail. Returns the
+  cook-expected-state-dict passed in."
+  [api-client cook-expected-state-dict pod]
+  (api/delete-pod api-client pod)
+  cook-expected-state-dict)
 
-(defn launch-task
-  [api-client expected-state-dict]
-  (api/launch-task api-client expected-state-dict)
-  expected-state-dict)
+(defn log-weird-state
+  "Weird. This pod is in a weird state. Log its weird state."
+  [cook-expected-state-dict k8s-actual-state-dict]
+  (log/error "Pod in a weird state:" cook-expected-state-dict "and k8s actual state" k8s-actual-state-dict))
+
+(defn kill-pod-weird
```

Comment:
Nitpick: Please rename to `kill-pod-in-weird-state`

Reply:
Done.